src/macros.jl

```diff
@@ -474,6 +498,59 @@ function _plural_macro_code(model, block, macro_sym)
     return code
 end
 
+function _string_summary(x)
+    if length(x) <= 75
+        return x
+    end
+    return x[1:32] * " [...] " * x[end-31:end]
+end
+
+"""
+    print_macro_timing_summary([io::IO = stdout], model::GenericModel)
+
+Print a summary of the runtime of each macro.
+
+## Example
+
+```jldoctest; filter=[r"Total time: .+ seconds", r"├.+", r"└.+"]
+julia> begin
+           model = Model()
+           @variable(model, x[1:2])
+           @objective(model, Min, sum(x))
+       end;
+
+julia> print_macro_timing_summary(model)
+Total time: 5.33690e-02 seconds
+│
+├ 2.96490e-02 s [55.55%]
+│ ├ REPL[8]:3
+│ └ `@variable(model, x[1:2])`
+│
+└ 2.37200e-02 s [44.45%]
+  ├ REPL[8]:4
+  └ `@objective(model, Min, sum(x))`
+```
+"""
+function print_macro_timing_summary(io::IO, model::GenericModel)
+    total_time = sum(values(model.macro_times))
+    times = sort!(collect(model.macro_times); by = last, rev = true)
+    println(io, "Total time: ", JuMP._format(total_time), " seconds")
```

Comment:
I like the way BenchmarkTools format times by using `ns` etc... when appropriate